core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/claim/DefaultClaimManager.java

```diff
@@ -111,18 +115,22 @@ public String getAttributeName(String domainName, String claimURI) throws UserSt
         if (claimCache.isInvalid()) {
             this.claimMapping = getClaimMapFromDB();
         }
-        ClaimMapping mapping = claimMapping.get(claimURI);
-        if (mapping != null) {
-            if (domainName != null) {
-                String mappedAttrib = mapping.getMappedAttribute(domainName.toUpperCase());
-                if (mappedAttrib != null) {
-                    return mappedAttrib;
+
+        if (claimURI != null) {
+            ClaimMapping mapping = claimMapping.get(claimURI);
+            if (mapping != null) {
+                if (domainName != null) {
+                    String mappedAttrib = mapping.getMappedAttribute(domainName.toUpperCase());
+                    if (mappedAttrib != null) {
+                        return mappedAttrib;
+                    }
+                    return mapping.getMappedAttribute();
+                } else {
+                    return mapping.getMappedAttribute();
                 }
-                return mapping.getMappedAttribute();
-            } else {
-                return mapping.getMappedAttribute();
             }
         }
+
         return null;
```

Comment:
Yes. I agree with your point. But returning "NULL" is the behavior we had already and there can be other dependent components which have implemented based on this behavior. So changing this would affect those components as well.